### ISSUE TITLE
systemd tmp file cleanup hang

### ISSUE THREAD

**Issue by RubenKelevra** [CachyOS/CachyOS-Settings#110]
Hi there!

I've been experiencing an issue where the systemd process for cleaning up temporary files occasionally hangs on my system. When it happens, one CPU core is fully utilized at 100% for an indefinite period, without any sign of completion.

Around the same time, I also started seeing a message about a duplicate configuration option during each package installation:

``` 
(3/4) Creating temporary files...
/usr/lib/tmpfiles.d/systemd.conf:35: Duplicate line for path "/var/lib/systemd/coredump", ignoring.
``` 

After checking the config files, it appears that there are two conflicting definitions for `/var/lib/systemd/coredump`:

``` 
❯ grep -R "/var/lib/systemd/coredump" /usr/lib/tmpfiles.d/ /etc/tmpfiles.d/ /run/tmpfiles.d/
/usr/lib/tmpfiles.d/systemd-tmp.conf:x  /var/lib/systemd/coredump/.#core*.%b*
/usr/lib/tmpfiles.d/systemd-tmp.conf:r! /var/lib/systemd/coredump/.#*
/usr/lib/tmpfiles.d/coredump.conf:d /var/lib/systemd/coredump 0755 root root 3d
/usr/lib/tmpfiles.d/systemd.conf:d /var/lib/systemd/coredump 0755 root root 2w
```

These definitions come from two different packages:

``` 
❯ pacman -F usr/lib/tmpfiles.d/coredump.conf
usr/lib/tmpfiles.d/coredump.conf is owned by cachyos/cachyos-settings 1:1.1.3-1

❯ pacman -F /usr/lib/tmpfiles.d/systemd.conf
usr/lib/tmpfiles.d/systemd.conf is owned by core/systemd 256.7-1
```

It seems one of these packages recently introduced this conflicting configuration, which may be linked to the CPU hang issue, though I can’t confirm if it’s the direct cause.

Just wanted to flag this as a potential packaging conflict. Thanks for taking a look!

**Comment by 1Naim** [comment]
It's not correlated. 

> (3/4) Creating temporary files...
> /usr/lib/tmpfiles.d/systemd.conf:35: Duplicate line for path "/var/lib/systemd/coredump", ignoring.

This is ignoring the duration set by us, and is not the cause of whatever hang you're experiencing. With that in mind, our setting is in fact useless here since what get used is systemd's settings instead. I'm not reading that wrong right? cc @ventureoo

**Comment by RubenKelevra** [comment]
@1Naim I did not change anything, but the hangs are gone. Guess an update fixed them.

**Comment by kxxt** [comment]
> This is ignoring the duration set by us, and is not the cause of whatever hang you're experiencing. With that in mind, our setting is in fact useless here since what get used is systemd's settings instead. I'm not reading that wrong right?

No, the duration set by CachyOS overrides the systemd's setting because `coredump.conf` comes first lexicographically when compared with `system.conf`.

Related: https://github.com/systemd/systemd/issues/8809

**Comment by 1Naim** [comment]
> > This is ignoring the duration set by us, and is not the cause of whatever hang you're experiencing. With that in mind, our setting is in fact useless here since what get used is systemd's settings instead. I'm not reading that wrong right?
> 
> No, the duration set by CachyOS overrides the systemd's setting because `coredump.conf` comes first lexicographically when compared with `system.conf`.
> 
> Related: [systemd/systemd#8809](https://github.com/systemd/systemd/issues/8809)

Reading the warning again, that makes sense. Thanks, this clears things up.

**Comment by nesper8** [comment]
what should i do to suppress the this?

im kind of lost

**Comment by DrCrinkle** [comment]
This was closed without fixing the warning. The CachyOS `coredump.conf` correctly overrides the retention to 3d (since it sorts before `systemd.conf` lexicographically), but the duplicate `d` line in `/usr/lib/tmpfiles.d/systemd.conf:35` still triggers a warning on every boot:

```
/usr/lib/tmpfiles.d/systemd.conf:35: Duplicate line for path "/var/lib/systemd/coredump", ignoring.
```

**Suggested fix:** Add a `.install` hook or pacman hook in `cachyos-settings` that installs a patched `/etc/tmpfiles.d/systemd.conf` shadowing the upstream one with line 35 (`d /var/lib/systemd/coredump 0755 root root 2w`) removed. This uses the standard tmpfiles.d override mechanism and cleanly suppresses the warning.

Alternatively, a simpler approach: ship the patched `systemd.conf` directly in the package under `/etc/tmpfiles.d/` so it shadows `/usr/lib/tmpfiles.d/systemd.conf` automatically.
